server/src/main/java/org/opensearch/cluster/service/MasterService.java

```diff
@@ -154,6 +159,20 @@ public MasterService(Settings settings, ClusterSettings clusterSettings, ThreadP
         );
         this.stateStats = new ClusterStateStats();
         this.threadPool = threadPool;
+        initializeMetrics(metricsRegistry);
+    }
+
+    private void initializeMetrics(MetricsRegistry metricsRegistry) {
+        this.clusterStateComputeHistogram = metricsRegistry.createHistogram(
+            "cluster.state.new.compute.latency",
+            "Histogram for recording time taken to compute new cluster state",
+            "ms"
+        );
+        this.clusterStatePublishHistogram = metricsRegistry.createHistogram(
+            "cluster.state.publish.success.latency",
```

Comment:
Does it include latency to push one node or all the nodes?

Reply:
This is the latency for successful publish of a cluster state. A state is said to be published successfully, once it receives the quorum of successful acks from other ClusterManager nodes. So, in a way, this metric includes latency to push to all the nodes.